### PR TITLE
Add telemetry around process payload encoding

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

Add telemetry around the process payloads encoding in the same fashion as the `jsonstream.bytes_in/out` of the core agent

```
# HELP process__payloads_bytes_in Count of bytes before encoding payload
# TYPE process__payloads_bytes_in counter
process__payloads_bytes_in{type="type:deployment"} 22548
process__payloads_bytes_in{type="type:replica-set"} 12408
process__payloads_bytes_in{type="type:service"} 9644
# HELP process__payloads_bytes_out Count of bytes after encoding payload
# TYPE process__payloads_bytes_out counter
process__payloads_bytes_out{type="type:deployment"} 6495
process__payloads_bytes_out{type="type:replica-set"} 4736
process__payloads_bytes_out{type="type:service"} 3734
```

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Relies on https://github.com/DataDog/agent-payload/pull/60

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/process/util/api/payload.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+// EncodePayload encodes a process message into a payload
+func EncodePayload(m model.MessageBody) ([]byte, error) {
+	msgType, err := model.DetectMessageType(m)
+	if err != nil {
+		return nil, fmt.Errorf("unable to detect message type: %s", err)
+	}
+
+	typeTag := "type:" + strconv.Itoa(int(msgType))
```
Comment (mfpierre) [inline pkg/process/util/api/payload.go:30]: might be worth to convert back this to a string 🤔 or we can learn https://github.com/DataDog/agent-payload/blob/6a8609f0fa8fb7ec6c5a9142bd459e1635e316bb/process/message.go#L75-L86 by heart 😬

File: pkg/process/util/api/payload.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+			Encoding: model.MessageEncodingZstdPB,
+			Type:     msgType,
+		}, Body: m})
+
+	tlmBytesOut.Add(float64(len(encoded)), typeTag)
+
+	return encoded, err
+}
+
+func messageTypeToString(m model.MessageType) string {
```
Comment (nammn) [inline pkg/process/util/api/payload.go:45]: It makes a lot of sense adding this. What do you think of moving this to agent-payload? 🤔 
https://github.com/DataDog/agent-payload/blob/master/process/message.go

Comment (mfpierre) [reply pkg/process/util/api/payload.go:45]: yeah it would make more sense to be in agent-payload, going to make the change

File: go.mod [hunk 3]
```diff
@@ -51,7 +51,7 @@ require (
 	code.cloudfoundry.org/rep v0.0.0-20200325195957-1404b978e31e // indirect
 	code.cloudfoundry.org/rfc5424 v0.0.0-20180905210152-236a6d29298a // indirect
 	code.cloudfoundry.org/tlsconfig v0.0.0-20200131000646-bbe0f8da39b3 // indirect
-	github.com/DataDog/agent-payload v0.0.0-20200731100214-6a8609f0fa8f // 4.38.0
+	github.com/DataDog/agent-payload v0.0.0-20200803141324-88e73ec658a0 // 4.38.0
```
Comment (mfpierre) [inline go.mod:54]: https://github.com/DataDog/agent-payload/pull/60 pending

**Review by kevinconaway — APPROVED** [review]

**Pull request merged**